src/HonzaBotner.Scheduler/Scheduler.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using HonzaBotner.Scheduler.Contract;
+
+namespace HonzaBotner.Scheduler
+{
+    public class SchedulerHostedService : BackgroundService
+    {
+        private readonly int _delay;
+        private readonly ILogger<SchedulerHostedService> _logger;
+        private readonly IList<CronJobWrapper> _cronJobs;
+
+        public SchedulerHostedService(int delay, IEnumerable<ICronJob> cronJobs, ILogger<SchedulerHostedService> logger)
+        {
+            DateTime now = DateTime.UtcNow;
+
+            _delay = delay;
+            _logger = logger;
+            _cronJobs = cronJobs
+                .Select(j => new CronJobWrapper(j, j.CronExpression, now))
+                .ToList();
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                DateTime currentTime = DateTime.UtcNow;
+
+               _logger.LogInformation("Scheduler running at: {Time}", currentTime.ToLocalTime());
+               await RunOnceAsync(currentTime, cancellationToken);
+
+               await Task.Delay(_delay, cancellationToken);
```

Comment:
Why this would not stop working after some time like we tried the `Task.Delay` before?

Reply:
This should be Ok. I dont remember exact reason why the previous usage of Task.Delay failed, but here, we have the main worker thread, which only does two thing, spawns new seperate threads and waits for the delay. No real work is done here.